src/Microsoft.Health.Fhir.Shared.Api/Modules/KnownAssemblies.cs

```diff
@@ -12,6 +12,6 @@ public static class KnownAssemblies
     {
         public static Assembly Core => typeof(Clock).Assembly;
 
-        public static Assembly CoreVersionSpecific => typeof(R4ModelInfoProvider).Assembly;
+        public static Assembly CoreVersionSpecific => typeof(VersionSpecificModelInfoProvider).Assembly;
```

Comment:
nit: I wonder if we should have some marker interface or something instead of using clock above. Someone might accidentally refactor the code and move Clock to say a utility library or something?